lib/documents.js

```diff
@@ -2266,9 +2267,39 @@ function writeAllDocumentsImpl(options) {
   operation.stream = inputStream;
   let result = requester.startRequest(operation);
   operation.response = result;
+  const writeBatchArray = [];
+  const batchSize = options.batchSize === null? 100:options.batchSize;
+
+  a:
+  for(let i=0; i<batchSize; i++) {
+    const record = inputStream.read();
+    if(record === null) {
+      switch(writeBatchArray.length) {
+        case 0: { finishWriter(options.writerId, done); break a;}
+        case batchSize: {
+          writeDocumentList.call(this, writeBatchArray, requestOptions);break a;}
+        default: {writeDocumentList.call(this, writeBatchArray, requestOptions).result((out) =>
+            finishWriter(writerId, done)).catch((err) => done(err));break a;}
+      }
+    }
+    writeBatchArray.push(record);
+  }
+
+  if(writeBatchArray.length !== 0) {
+    writeDocumentList.call(this, writeBatchArray, requestOptions).result((output) => writeAllDocumentsImpl(writerId, done))
+        .catch((err) => console.log(err));
+  }
+
   return inputStream;
 }
 
+function finishWriter(writerId, done) {
```

Comment:
This function should take the operation object instead of done and should access the requesterCount as a property of the operation object.